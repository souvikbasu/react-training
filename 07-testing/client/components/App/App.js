import React from 'react';

export default class App extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      number: 4
    }
  }

  render() {
    return (
      <div style={{textAlign: 'center'}} className="App" >
        <span>Number : </span>
        <span className="number">{this.state.number}</span>
      </div>);
  }
}
