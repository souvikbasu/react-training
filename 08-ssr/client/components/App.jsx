import React from 'react';

export default class App extends React.Component {
  render() {
    return (
      <div style={{textAlign: 'center'}}>
        <h1>This is React Component</h1>
      </div>);
  }
}
