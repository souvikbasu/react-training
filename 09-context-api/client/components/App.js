import React from 'react';

let basketCount = 1;

const BasketContext = React.createContext();

class App extends React.Component {
  render() {
    return (
      <BasketContext.Provider value={basketCount} >
        <Header />
      </BasketContext.Provider>
    );
  }
}

function Header(props) {
  return (
    <div>
      <BasketIcon />
    </div>
  );
}

function BasketIcon(props) {
  return (
    <BasketContext.Consumer>
      { basketCount => 
        (
          <div>
            <img src="../images/basket.png" />
            <div>{basketCount}</div>
          </div>
        )
      }
    </BasketContext.Consumer>
  );
}

export default App;