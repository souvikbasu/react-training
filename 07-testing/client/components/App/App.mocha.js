import React from 'react';
import App from './App';
import {shallow} from 'enzyme';

const container = shallow(<App />);

describe('App component', () => {

  it('Should contain App class', () => {
    expect(container.find('.App').length).to.equal(1);
  });

  it('Should have number set to 1', () => {
    expect(container.find('.number').text()).to.equal("1");
  });
});
