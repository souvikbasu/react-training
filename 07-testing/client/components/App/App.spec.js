import React from 'react';
import App from './App';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer
    .create(<App />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});