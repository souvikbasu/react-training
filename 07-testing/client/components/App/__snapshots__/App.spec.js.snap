// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="App"
  style={
    Object {
      "textAlign": "center",
    }
  }
>
  <span>
    Number : 
  </span>
  <span
    className="number"
  >
    4
  </span>
</div>
`;
